token/core/common/meta/metadata.go

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package meta
+
+import (
+	"strings"
+)
+
+const (
+	TransferMetadataPrefix = "TransferMetadataPrefix"
+)
+
+// SetTransferAction extracts the transfer metadata from the passed attributes and
+// sets them to the passed metadata
+func SetTransferAction(attrs map[interface{}]interface{}, metadata map[string][]byte) {
+	for key, value := range attrs {
```

Comment:
Modifying an input parameter is confusing for the logic flow. Instead I would return a new map:
```
func GetTransferActions(attrs map[any]any) map[string][]byte {
metadata := map[string][]byte
for ...
return metadata
}
```
and then loop through the map and set the values. The big O complexity is the same.

Reply:
👍